### PR TITLE
Move blog to static website

### PR DESCRIPTION
**Pull request by patersonc**
Move blog to static website from the foundation website.

Part of https://github.com/kernelci/kernelci-project/issues/43

### CODE REVIEW

**Comment by patersonc** [comment]
Keeping in draft for now as I'll add some commits to bring the current blog posts over from the foundation site.

File: kernelci.org/config.toml [hunk 1]
```diff
@@ -85,8 +85,3 @@ footer_about_disable = false
     identifier = "lf"
     name = "LF"
     url = "https://foundation.kernelci.org"
-
-  [[menu.main]]
```
Comment (gctucker) [inline kernelci.org/config.toml:89]: I think we should keep the Blog menu entry, just drop the `url` as it'll be using the standard one from the Docsy theme.  See this example website which features a blog section:
https://example.docsy.dev/

**Comment by gctucker** [comment]
@patersonc It would be a bit cleaner to use a fork under your own GitHub user account when creating PRs rather than pushing branches to the main repository.  These are visible to anyone making a clone of it.

**Comment by gctucker** [comment]
This PR conflicts with #49 due to the changes in the main menu.  So #49 could have the `staging-skip` label for now to get the blog changes from this PR deployed on the staging static website.

**Comment by patersonc** [comment]
> @patersonc It would be a bit cleaner to use a fork under your own GitHub user account when creating PRs rather than pushing branches to the main repository. These are visible to anyone making a clone of it.

Ah, sorry! Shall I close my current MRs and start over?

File: kernelci.org/config.toml (hunk 1 above)
Comment (patersonc) [reply kernelci.org/config.toml:89]: The current implementation would result in two headings, "News" and "Posts".

image

Happy to merge into a single "Blog" entry if you'd prefer that.

**Comment by gctucker** [comment]
> > @patersonc It would be a bit cleaner to use a fork under your own GitHub user account when creating PRs rather than pushing branches to the main repository. These are visible to anyone making a clone of it.
> 
> Ah, sorry! Shall I close my current MRs and start over?

No it's not worth doing that, just something to keep in mind for future PRs I guess.

File: kernelci.org/config.toml (hunk 1 above)
Comment (patersonc) [reply kernelci.org/config.toml:89]: I'll merge the two categories into one.

Comment (patersonc) [reply kernelci.org/config.toml:89]: I've kept the two categories, but changed it so we have a single "Blog" link on the main menu

**Comment by patersonc** [comment]
I'm removing the draft label as all blog content from the foundation site has now been ported.

**Comment by gctucker** [comment]
> I'm removing the draft label as all blog content from the foundation site has now been ported.

Now I'm starting a blog post about KernelCI hackfests on top of this ;)

**Comment by gctucker** [comment]
BTW I don't think we actually need the hierarchy with all the months and days, the date is collected from the front matter in each post.

**Review by gctucker — COMMENTED** [review]
Please use git-lfs for the binary files, it makes it much easier to do a lightweight checkout as this repository is not only for the static website.

**Comment by patersonc** [comment]
> Please use git-lfs for the binary files, it makes it much easier to do a lightweight checkout as this repository is not only for the static website.

Done. Hopefully.

**Comment by patersonc** [comment]
> BTW I don't think we actually need the hierarchy with all the months and days, the date is collected from the front matter in each post

I've got mixed feelings about this.

Whilst I agreed the hierarchy isn't essential as our output is low, I think having no hierarchy would become a bit of a mess.
Perhaps there is another directory structure we could choose, but I can't think of anything more suitable.

**Comment by gctucker** [comment]
> > BTW I don't think we actually need the hierarchy with all the months and days, the date is collected from the front matter in each post
> 
> I've got mixed feelings about this.
> 
> Whilst I agreed the hierarchy isn't essential as our output is low, I think having no hierarchy would become a bit of a mess.
> Perhaps there is another directory structure we could choose, but I can't think of anything more suitable.

How about just directories for each year?

**Comment by gctucker** [comment]
> > Please use git-lfs for the binary files, it makes it much easier to do a lightweight checkout as this repository is not only for the static website.
> 
> Done. Hopefully.

Erm, there's a problem when checking out your branch:
```
HEAD is now at 33632cc8b4ac kernelci.org: post: Add "The first ever KernelCI hackfest" blog post
Encountered 13 file(s) that should have been pointers, but weren't:
	kernelci.org/content/en/blog/news/2020/08/19/kernelci-blog.png
	kernelci.org/content/en/blog/posts/2020/05/11/featured-image-blog2.png
	kernelci.org/content/en/blog/posts/2020/07/09/ci-system-usage-1024px.png
	kernelci.org/content/en/blog/posts/2020/07/09/kernelci-community-survey-report-1200x325-1.png
	kernelci.org/content/en/blog/posts/2020/07/09/reporting-1024px.png
	kernelci.org/content/en/blog/posts/2020/07/09/response-time-1024px.png
	kernelci.org/content/en/blog/posts/2020/07/09/roles-distribution-1024px.png
	kernelci.org/content/en/blog/posts/2020/07/09/upstream-contributions-1024px.png
	kernelci.org/content/en/blog/posts/2020/08/21/kcidb_object_relations.svg
	kernelci.org/content/en/blog/posts/2020/08/21/kcidb_object_submitting.gif
	kernelci.org/content/en/blog/posts/2020/08/21/patched_revision2.png
	kernelci.org/content/en/blog/posts/2020/08/21/revision1.png
	kernelci.org/content/en/blog/posts/2021/03/16/looking-back-looking-forward.png
```
For each binary file, you need to do `git-lfs track` and then `git add`.  I guess you need to make a copy of the files and do `git rm` first on all of them, commit that to not have the binaries in git, then add them again with git-lfs.  Let me know if this is too confusing and I'll help you out.

**Comment by gctucker** [comment]
The git-lfs issue is causing errors on staging, so adding the `staging-skip` label again for now.

**Comment by gctucker** [comment]
Replaced with #52, the image files had to be added with `git add` as well as `git-lfs track`.

**Comment by patersonc** [comment]
> Replaced with #52, the image files had to be added with `git add` as well as `git-lfs track`.

Oh, I thought I did do that. Maybe I missed one?

**Comment by patersonc** [comment]
>How about just directories for each year?

Could do. Let me know if you want me to make the change.

**Comment by gctucker** [comment]
PR #52 has all your changes with the "git add" fix.  So I think we can close this PR #50 and get #52 merged instead, or you could take the commits from #52 and force-push to the branch used in #50 and then we'll close #52.

**Comment by patersonc** [comment]
> PR #52 has all your changes with the "git add" fix. So I think we can close this PR #50 and get #52 merged instead, or you could take the commits from #52 and force-push to the branch used in #50 and then we'll close #52.

Just close this one and delete my branch :)

**Comment by gctucker** [comment]
Alright then :)

**Pull request closed without merging**